File: 06-06-2025/QuizupAPI/Models/Student.cs
using System.ComponentModel.DataAnnotations;
using QuizupAPI.Interfaces;

namespace QuizupAPI.Models
{
    public class Student : ISoftDeletable
    {
        public long Id { get; set; }
        [Required]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        public string LastName { get; set; } = string.Empty;

        public Class? Class { get; set; }
        public long ClassId { get; set; }

        public bool IsDeleted { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        public User? User { get; set; }
        
        public ICollection<QuizSubmission>? QuizSubmissions { get; set; }
    }
}

File: Quizup-Angular-Application/src/app/features/student/models/student.ts
import { ClassModel } from '../../class/models/class';
import { QuizSubmission } from '../../quiz/models/quizSubmission.model';

export class StudentModel {
  constructor(
    public id: number = 0,
    public email: string = '',
    public firstName: string = '',
    public lastName: string = '',
    public classe: ClassModel = new ClassModel(),
    public createdAt: Date = new Date(),
    public quizSubmissions?: QuizSubmission[]
  ) {}
}
